base/linalg/exceptions.jl

```diff
@@ -29,10 +29,30 @@ function Base.showerror(io::IO, ex::ARPACKException)
     end
 end
 
+"""
+   SingularException(info)
+
+The matrix passed is a singular matrix. Optional argument INFO is an INTEGER > 0  
+      if INFO = i, The matrix is exactly singular, and division by zero 
+      will occur if it is used to solve a system of equations.
+"""
 struct SingularException <: Exception
     info::BlasInt
 end
 
+function Base.showerror(io::IO, ex::SingularException)
+    print(io, "SingularException($ex.info): matrix is singular")
+end
+
+"""
+    PosDefException(info)
+
+Optional Argument info is an Integer. 
+If info == -1: The matrix is not Hermitian.
+Else: The matrix is not positive definite.
```

Comment:
This is close but not quite correct since a zero value would mean that the matrix is PD. It would probably also be good to mention that this was tested by a Cholesky factorization, i.e. something like "if `info>0` then the Cholesky factorization failed at the $(info) diagonal element which indicates that the matrix is not positive definite."